packages/server-wallet/jest/with-peers-setup-teardown.ts

```diff
@@ -0,0 +1,69 @@
+import {Wallet} from '../src/wallet';
+import {DBAdmin, defaultTestConfig, overwriteConfigWithDatabaseConnection} from '../src';
+import {
+  seedAlicesSigningWallet,
+  seedBobsSigningWallet,
+} from '../src/db/seeds/1_signing_wallet_seeds';
+import {Participant} from '@statechannels/client-api-schema';
+import {makeDestination} from '@statechannels/wallet-core';
+
+interface TestPeerWallets {
+  a: Wallet;
+  b: Wallet;
+}
+
+export const aWalletConfig = overwriteConfigWithDatabaseConnection(defaultTestConfig(), {
+  database: 'TEST_A',
+});
+export const bWalletConfig = overwriteConfigWithDatabaseConnection(defaultTestConfig(), {
+  database: 'TEST_B',
+});
+export let participantA: Participant;
+export let participantB: Participant;
+export let peerWallets: TestPeerWallets;
+
+export function getPeersSetup(withWalletSeeding: boolean = false): jest.Lifecycle {
+  return async () => {
+    await Promise.all([DBAdmin.dropDatabase(aWalletConfig), DBAdmin.dropDatabase(bWalletConfig)]);
+
+    await Promise.all([
+      DBAdmin.createDatabase(aWalletConfig),
+      DBAdmin.createDatabase(bWalletConfig),
+    ]);
+
+    await Promise.all([
+      DBAdmin.migrateDatabase(aWalletConfig),
+      DBAdmin.migrateDatabase(bWalletConfig),
+    ]);
+
+    peerWallets = {
+      a: await Wallet.create(aWalletConfig),
+      b: await Wallet.create(bWalletConfig),
+    };
+
+    if (withWalletSeeding) {
+      await seedAlicesSigningWallet(peerWallets.a.knex);
+      await seedBobsSigningWallet(peerWallets.b.knex);
```

Comment:
Suppose someone calls `getPeersSetup(false)`. Is there a reason _not_ to seed these? (Do existing tests depend on these not being seeded??)

Reply:
(from memory) As is, four of the five tests do call `getPeersSetup(false)`.

Will investigate - intent was to faithfully reproduce initial conditions on existing tests, which I could do (without knowing anything about wallets or how the tests used them) by adding this flag.

Reply:
That approach is great, and makes this easy to review! It might have been worth a comment in the commit message or in this file.

 IMO doing it here is scope creep -- if you are planning on changing the beforeAll usage, it should be in a follow up PR.

Reply:
I disagree with that. If there is a cost of changing a unit test file in order to create a more standardized way of using `beforeAll`/`afterAll`, I think it is totally fine to do so in this PR even if there is scope creep.

That being said, a valid reason to avoid changing the behavior of the unit test file is to merge the PR quicker.